kasa/interfaces/thermostat.py

```diff
@@ -0,0 +1,96 @@
+"""Interact with a TPLink Thermostat."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Literal
+
+from ..module import Module
+
+
+class ThermostatState(Enum):
+    """Thermostat state."""
+
+    Heating = "heating"
+    Calibrating = "progress_calibration"
+    Idle = "idle"
+    Off = "off"
+    Unknown = "unknown"
+
+
+class Thermostat(Module, ABC):
+    """Base class for TP-Link Thermostat."""
+
+    @property
+    @abstractmethod
+    def state(self) -> bool:
+        """Return thermostat state."""
+
+    @abstractmethod
+    async def set_state(self, enabled: bool):
+        """Set thermostat state."""
+
+    @property
+    @abstractmethod
+    def mode(self) -> ThermostatState:
+        """Return thermostat state."""
+
+    @property
+    @abstractmethod
+    def allowed_temperature_range(self) -> tuple[int, int]:
+        """Return allowed temperature range."""
+
+    @property
+    @abstractmethod
+    def minimum_target_temperature(self) -> int:
+        """Minimum available target temperature."""
+
+    @property
+    @abstractmethod
+    def maximum_target_temperature(self) -> int:
+        """Minimum available target temperature."""
+
+    @property
+    @abstractmethod
+    def target_temperature(self) -> float:
+        """Return target temperature."""
+
+    @abstractmethod
+    async def set_target_temperature(self, target: float):
+        """Set target temperature."""
+
+    @property
+    @abstractmethod
+    def temperature_offset(self) -> int:
+        """Return temperature offset."""
+
+    @abstractmethod
+    async def set_temperature_offset(self, offset: int):
+        """Set temperature offset."""
```

Comment:
We should only expose primary controls over the common interafce.

Reply:
Removed